### PR TITLE
Update the extention tutorial in ckan document

### LINKED ISSUE #6632: Writing extensions tutorial is not up to date
**CKAN version**
2.9

**Describe the bug**
[This](https://docs.ckan.org/en/2.9/extensions/tutorial.html#creating-a-plugin-class) and [that](https://docs.ckan.org/en/2.9/extensions/tutorial.html#adding-the-plugin-to-setup-py) are not needed anymore.

### PR DESCRIPTION
**Pull request by Sunny-NEC**
Fixes #6632 

### Features:

- [ ] includes tests covering changes
- [x ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by wardi** [comment]
Please fix the new warning from the sphinx build:
```
/root/project/doc/extensions/tutorial.rst:140: WARNING: undefined label: setup.py (if the link has no caption the label must precede a section header)
```

File: doc/extensions/tutorial.rst [hunk 1]
```diff
@@ -137,7 +137,7 @@ directory, and run ``python setup.py develop``:
 Enabling the plugin
 ===================
 
-An extension's plugins must be added to the :ref:`ckan.plugins` setting in your
+An extension's plugins must be added to the setting in your
```
Comment (wardi) [inline doc/extensions/tutorial.rst:140]:
```suggestion
An extension's plugins must be added to the ``ckan.plugins`` setting in your
```

**Comment by Gauravp-NEC** [comment]
@wardi , What is left to do in this PR?

**Comment by Gauravp-NEC** [comment]
@wardi , What is left to do in this PR?

**Comment by Gauravp-NEC** [comment]
@wardi , **Sunny-NEC**  is no loner working on this PR. Please let me know if anything is left to do in this PR, I will do the rework. If this PR looks good then please merge it. Thanks :)

**Comment by wardi** [comment]
@kowh-ai does removing these sections make sense?

**Comment by amercader** [comment]
I don't think we need to remove them. Using `ckan generate` means you don't have to do it manually, but it is still worthwhile documenting all the steps. Maybe we can add a `..note::` explaining this.

The whole tutorials section should be reviewed to adapt it to modern practices.

**Comment by robin-NEC** [comment]
Hi @wardi  @amercader, I would like to do the rework for this PR. 
According to my understanding these sections are still valid if we are using `cookiecutter` instead of `ckan generate`.
Please correct me if I'm wrong.

**Comment by robin-NEC** [comment]
> I don't think we need to remove them. Using `ckan generate` means you don't have to do it manually, but it is still worthwhile documenting all the steps. Maybe we can add a `..note::` explaining this.
> 
> The whole tutorials section should be reviewed to adapt it to modern practices.

@amercader based on your suggestion I've proposed the changes on https://github.com/ckan/ckan/pull/6808

**Pull request closed without merging**